packages/Checkbox/src/Checkbox.js

```diff
@@ -0,0 +1,69 @@
+import React from "react";
+import PropTypes from "prop-types";
+import uuid from "uuid/v4";
+import { ShirtSizes } from "@paprika/helpers/lib/customPropTypes";
+import CheckIcon from "@paprika/icon/lib/Check";
+import AddIcon from "@paprika/icon/lib/Add";
+import checkboxStyles from "./Checkbox.styles";
+
+const propTypes = {
+  a11yText: PropTypes.string,
+  children: PropTypes.node,
+  isChecked: PropTypes.bool,
+  isDisabled: PropTypes.bool,
+  isIndeterminate: PropTypes.bool,
+  onChange: PropTypes.func.isRequired,
+  size: PropTypes.oneOf(ShirtSizes.DEFAULT),
+};
+
+const defaultProps = {
+  a11yText: null,
+  children: null,
+  isChecked: false,
+  isDisabled: false,
+  isIndeterminate: false,
+  size: ShirtSizes.MEDIUM,
+};
+
+const Checkbox = props => {
+  const checkboxId = React.useRef(uuid());
+  const inputRef = React.useRef(null);
+
+  const { a11yText, children, isChecked, isDisabled, isIndeterminate, size, ...moreProps } = props;
+
+  const styleProps = {
+    hasChildren: !!children,
+    size,
+  };
+
+  React.useEffect(() => {
+    if (inputRef.current) inputRef.current.indeterminate = isIndeterminate;
+  }, [isIndeterminate]);
```

Comment:
This should also be dependent on `isChecked`.

Reply:
Actually, on second though, this might be problematic too... I think it might work better if there was a `checkedState: oneOf(["checked", "unchecked", "indeterminate"])` or something like that, since the states are all mutually exclusive.